codes/README.md

```diff
@@ -1,66 +1,138 @@
 # Code Framework
-The overall code framework is shown in the following figure. It mainly consists of four parts - `Config`, `Data`, `Model` and `Network`.
+
+The overall code framework is shown in the following figure. It mainly consists of four parts - [Config], [Data],
+[Model], and [Network].
 
 <p align="center">
    <img src="https://github.com/xinntao/public_figures/blob/master/BasicSR/code_framework.png" height="450">
 </p>
 
-Let us take the train commad `python train.py -opt options/train/train_esrgan.json` for example. A sequence of actions will be done after this command. 
+For example, once [train.py] is called (i.e. `python train.py -i options/train/train_template.yml`):
+
+1.  [Config] - Reads the configuration from [/options/train/train_template.yml] which is a [YAML] file. Then passes
```

Comment:
It is mentioned as an alternative in the [Config] section. It specifically states YAML since the example file given is in fact a YAML file. If they click [Config] to read more info, they can learn about JSON.

Reply:
in my opinion, the JSON configs should just be removed. They are an objectively worse option

Reply:
I agree, there's no way vic wants to deal with coding those dodgy comment-supported JSON's lmao

Reply:
Many people prefer JSON, some even .ini files or python dictionaries.